### PR TITLE
Add inf & nan tests for test_constraints_func

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
`test_constraints_func` in `test_nsgaii.py` lacks test cases for `inf` or `nan` values. We add those test cases.

## Description of the changes
Add `inf` and `nan` cases for `test_constraints_func`.

### CODE REVIEW

**Comment by contramundum53** [comment]
@not522 @knshnb Could you review this PR?

**Review by not522 — APPROVED** [review]
LGTM!

File: tests/samplers_tests/test_nsgaii.py [hunk 1]
```diff
@@ -129,7 +138,7 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
         nonlocal constraints_func_call_count
         constraints_func_call_count += 1
 
-        return (trial.number,)
+        return (constraint_value,)
```
Comment (knshnb) [inline tests/samplers_tests/test_nsgaii.py:141]: In the current code, the case where `constraint_value` is different between trials is not tested, which was originally tested.
How about returning something like `(constraint_value + trial.number,)`?

**Review by knshnb — COMMENTED** [review]
Thanks for enhancing the test cases. I left a small comment. PTAL.

File: tests/samplers_tests/test_nsgaii.py (hunk 1 above)
Comment (contramundum53) [reply tests/samplers_tests/test_nsgaii.py]: Thanks for pointing that out! I fixed the code. PTAL.

**Review by knshnb — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**